specs/order_spec.rb

```diff
@@ -75,35 +79,204 @@
+    #   - Order.all returns an array
+    #   - Everything in the array is an Order
+    #   - The number of orders is correct
+    #   - The ID and products of the first and last
+    #       orders match what's in the CSV file
+    # Feel free to split this into multiple tests if needed
     it "Returns an array of all orders" do
-      # TODO: Your test code here!
-      # Useful checks might include:
-      #   - Order.all returns an array
-      #   - Everything in the array is an Order
-      #   - The number of orders is correct
-      #   - The ID and products of the first and last
-      #       orders match what's in the CSV file
-      # Feel free to split this into multiple tests if needed
-    end
-  end
+      Grocery::Order.all.must_be_kind_of Array
+    end # it "Returns an array of all orders" do
+
+    it "will have erything in the Order.all array be an instsnce of the Order class" do
+      test = Grocery::Order.all
+      test.length.times do |i|
+        test[i].must_be_kind_of Grocery::Order
+      end
+    end #it "will have erything....
+
+    it "will contain the right numnber of orders" do
+      all_orders = []
+      CSV.open("support/orders.csv", 'r').each do |line|
+        all_orders << Grocery::Order.new(line[0], line[1].split(";"))
+      end
+
+      Grocery::Order.all.length.must_equal all_orders.length
+    end #it "will contain ....
+
+    it "will have the right id for the first order" do
+      first_order = CSV.open('support/orders.csv', 'r') { |csv| csv.first }
+      test = Grocery::Order.all
+
+      test[0].id.must_equal first_order[0][0]
+    end #it "will have the right id for the first order
+
+    it "will have the right product for the first order" do
+      first_order = CSV.open('support/orders.csv', 'r') { |csv| csv.first }
+      order_products = {}
+      products = first_order[1].split(";")
+      products.each do |item_price|
+        product_then_price = item_price.split(":")
+        order_products[product_then_price[0]] = product_then_price[1].to_f
+      end
+      test_hash = {first_order[0] => order_products}
+
+      test = Grocery::Order.all
+
+      test[0].products.must_equal test_hash[first_order[0]]
+    end #it "will have the right product for the first order
+
+
+    it "will have the right id for the last order" do
+      #TODO is there a better way to do this that does't basically repeat the code in order.rb?
+      all_orders = []
+      CSV.open("support/orders.csv", 'r').each do |line|
+        all_orders << Grocery::Order.new(line[0], line[1].split(";"))
+      end
+
+      test = Grocery::Order.all
+      test[-1].id.must_equal all_orders[-1].id.to_s
+    end #it "will have the right id for the last order
+
+    it "will have the right products for the last order" do
+      #TODO is there a better way to do this that does't basically repeat the code in order.rb?
+      all_orders = []
+      CSV.open("support/orders.csv", 'r').each do |line|
+        id = line[0]
```

Comment:
While I appreciate the idea of going to the source, reimplementing all this logic to read the CSV file is a little clunky. It might be more effective to just hard-code the expected results.

If you definitely want to check it against the CSV, maybe you could write a helper method for your tests that pulls out the product list for order X.